framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -7341,6 +7487,14 @@ VkResult VulkanReplayConsumerBase::OverrideCreateRayTracingPipelinesKHR(
     VkDeferredOperationKHR                   in_deferredOperation =
         (deferred_operation_info != nullptr) ? deferred_operation_info->handle : VK_NULL_HANDLE;
     VkPipelineCache in_pipelineCache = (pipeline_cache_info != nullptr) ? pipeline_cache_info->handle : VK_NULL_HANDLE;
+    VkPipelineCache overridePipelineCache = in_pipelineCache;
+
+    // If there is no pipeline cache and we want to create a new one
+
+    if (in_pipelineCache == VK_NULL_HANDLE && options_.add_new_pipeline_caches)
+    {
+        overridePipelineCache = CreateNewPipelineCache(device_info, *pPipelines->GetPointer());
```

Comment:
`--add-new-pipeline-cache` only adds a cache to pipelines which do not use one. The cache can be saved with `--save-pipeline-cache` but it wont be used by `--load-pipeline-cache` because when `--add-new-pipeline-cache` is set, a new cache is created while when it is not, no cache is provided since the captured command has a null handle. At least, that is my understanding of the code - I could be wrong about this.

I believe `--add-new-pipeline-cache` should switch behavior based on whether --save-pipeline-cache is specified where a single cache is created and used for all 'null cache' pipelines, then when --load-pipeline-cache the code looks through the loaded file for a cache entry which corresponds to the 'null cache' pipelines and uses that. This way, the caching can still occur.

Reply:
Let's push it to later for now.  Create a new issue.